exercises/error-handling/src/lib.rs

```diff
@@ -0,0 +1,40 @@
+use std::cmp;
+
+#[derive(Eq,PartialEq,Debug)]
+pub enum IceCream {
+    Chocolate,
+    Vanilla,
+    Strawberry,
+}
+
+#[derive(Debug)]
+pub struct IceCreamTruck {
+    // These fields are only pub, because tests are not in the same file
+    // and we need to verify inventory after actions.
+    pub chocolate: u32,
```

Comment:
this representation is a little unfortunate when considering the fact that `IceCream` exists above. One might expect we would have map of `IceCream` to quantity, but this is no longer possible given the struct must be of this form.

as an alternative, might imagine a function that queries the ice cream truck for how many of each flavour it has left, perhaps?

Then the fields would not need to be `pub`, and in fact the student may choose any internal representation.